### ISSUE TITLE
Error during AirPrice request.

### ISSUE THREAD

**Issue by shmuga** [Travelport-Ukraine/uapi-json#69]
Next code is generating `000280 FLIGHT NOT FOUND - CHECK AVAILABILITY`. 
```xml
<?xml version="1.0" encoding="utf-8"?>
<!--Release 8.1-->
<!--Version Dated as of 15/Apr/2015 11:24:06-->
<!--Air Pricing For Galileo(1G) with LFS CheckFlightDetails Request-->
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
	<soap:Header/>
	<soap:Body>
		<air:AirPriceReq xmlns:air="http://www.travelport.com/schema/air_v36_0" AuthorizedBy="user" CheckFlightDetails="true" TargetBranch="BRANCH" TraceId="">
			<com:BillingPointOfSaleInfo xmlns:com="http://www.travelport.com/schema/common_v36_0" OriginApplication="UAPI"/>
			<air:AirItinerary>
				<air:AirSegment ArrivalTime="2016-11-08T12:30:00.000+04:00" DepartureTime="2016-11-08T08:45:00.000+02:00" Carrier="YE" CabinClass="Economy" Origin="IEV" Destination="TBS" ETicketability="Yes" Equipment="733" FlightNumber="211" LinkAvailability="true" PolledAvailabilityOption="Polled avail exists" ProviderCode="1G" Key="0" Group="0">
</air:AirSegment>
			</air:AirItinerary>
			<air:AirPricingModifiers InventoryRequestType="Direct">
				<air:PermittedCabins>
					<com:CabinClass xmlns:com="http://www.travelport.com/schema/common_v36_0" Type="Economy"/>
				</air:PermittedCabins>
			</air:AirPricingModifiers>
			<com:SearchPassenger xmlns:com="http://www.travelport.com/schema/common_v36_0" Key="P_0" Code="ADT" Age="54"/>
		</air:AirPriceReq>
	</soap:Body>
</soap:Envelope>
```

**Comment by Smotrov** [comment]
Is this flight is really available in terminal? Please check A or FS.

**Comment by shmuga** [comment]
The cause of problem is that there is no such airline anymore and screens of `A` and `FS` return different  arrival/departure time. So AirPrice returns ` 000280 FLIGHT NOT FOUND - CHECK AVAILABILITY.`

`FS`
![screen shot 2016-11-04 at 13 38 17](https://cloud.githubusercontent.com/assets/4268223/20004534/9eb6f9a4-a295-11e6-909f-2e4d56a9ce21.png)

`A`
![screen shot 2016-11-04 at 13 38 01](https://cloud.githubusercontent.com/assets/4268223/20004544/abf2c648-a295-11e6-8cc9-03dffda8707f.png)

**Comment by shmuga** [comment]
Huston, we've got problems again.
```xml
<?xml version="1.0" encoding="utf-8"?>
<!--Release 8.1-->
<!--Version Dated as of 15/Apr/2015 11:24:06-->
<!--Air Pricing For Galileo(1G) with LFS CheckFlightDetails Request-->
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
	<soap:Header/>
	<soap:Body>
		<air:AirPriceReq xmlns:air="http://www.travelport.com/schema/air_v36_0" AuthorizedBy="user" CheckFlightDetails="true" TargetBranch="P2660493" TraceId="">
			<com:BillingPointOfSaleInfo xmlns:com="http://www.travelport.com/schema/common_v36_0" OriginApplication="UAPI"/>
			<air:AirItinerary>
				<air:AirSegment ArrivalTime="2016-11-23T13:45:00-05:00" DepartureTime="2016-11-23T10:05:00+02:00" Carrier="PS" CabinClass="Economy" Origin="KBP" Destination="JFK" ETicketability="Yes" Equipment="767" FlightNumber="231" LinkAvailability="true" PolledAvailabilityOption="Polled avail exists" ProviderCode="1G" Key="0" Group="0">
</air:AirSegment>
				<air:AirSegment ArrivalTime="2016-12-26T17:20:00+02:00" DepartureTime="2016-12-26T00:30:00-05:00" Carrier="PS" CabinClass="Economy" Origin="JFK" Destination="KBP" ETicketability="Yes" Equipment="767" FlightNumber="232" LinkAvailability="true" PolledAvailabilityOption="Polled avail exists" ProviderCode="1G" Key="1" Group="0">
</air:AirSegment>
			</air:AirItinerary>
			<air:AirPricingModifiers InventoryRequestType="Direct">
				<air:PermittedCabins>
					<com:CabinClass xmlns:com="http://www.travelport.com/schema/common_v36_0" Type="Economy"/>
				</air:PermittedCabins>
			</air:AirPricingModifiers>
			<com:SearchPassenger xmlns:com="http://www.travelport.com/schema/common_v36_0" Key="P_0" Code="ADT" Age="33"/>
			<com:SearchPassenger xmlns:com="http://www.travelport.com/schema/common_v36_0" Key="P_1" Code="ADT" Age="34"/>
		</air:AirPriceReq>
	</soap:Body>
</soap:Envelope>
```

During this request we are receiving such error 
```xml
<?xml version="1.0" encoding="utf-8"?>
<SOAP:Envelope xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/">
	<SOAP:Body>
		<SOAP:Fault>
			<faultcode>Server.System</faultcode>
			<faultstring>NO FARES AVAILABLE FOR REQUESTED CABIN</faultstring>
			<detail>
				<common_v36_0:ErrorInfo xmlns:common_v36_0="http://www.travelport.com/schema/common_v36_0">
					<common_v36_0:Code>-1</common_v36_0:Code>
					<common_v36_0:Service>AirPriceService</common_v36_0:Service>
					<common_v36_0:Type>System</common_v36_0:Type>
					<common_v36_0:Description>NO FARES AVAILABLE FOR REQUESTED CABIN</common_v36_0:Description>
					<common_v36_0:TransactionId>3F9393130A0759C4BB2200130406AEA4</common_v36_0:TransactionId>
					<common_v36_0:TraceId/>
				</common_v36_0:ErrorInfo>
			</detail>
		</SOAP:Fault>
	</SOAP:Body>
</SOAP:Envelope>
```

But in terminal `A` screens are next:

![screen shot 2016-11-07 at 18 07 25](https://cloud.githubusercontent.com/assets/4268223/20065927/892181ce-a518-11e6-9786-c706491d148c.png)

![screen shot 2016-11-07 at 18 07 13](https://cloud.githubusercontent.com/assets/4268223/20065934/8d8708ec-a518-11e6-987d-03af6a184caf.png)


@mbrandysky could help us with this one, please?

**Comment by mbrandysky** [comment]
I would say that the transaction is absolutelly correct and for PremiumEconomy and Business it works properly. If I remove cabin specification, the system return combination of Economy/PremiumEconomy. But In my testing PCC is RBD "L" available, it is in economy and via Smartpoint Im able to book it without any problem. L is in Economy according to the IATA, Fare rules, there is no GDS exception. I have to report this issue on apisupport and also give me more time today for research.

**Comment by mbrandysky** [comment]
I have to wait for apisupport. I found two problems in the request but none of them causes this problem.

- Value of Group in AirSegment should respect journey (for a basic return trip it should be Group="0" for journey there and Group="1" for journey back).
- Value of InventoryRequestType should be "DirectAccess" instead of "Direct".

**Comment by mbrandysky** [comment]
I have a response from apisupport, that they are unable to recreate the situation. For future dates the Galileo is able to return offers including same problematic flights :-(

**Comment by mbrandysky** [comment]
But my remarks about the necessary changes are still valid.

**Comment by shmuga** [comment]
Recommendations implemented in the PR  #80
